packages/yarnpkg-core/sources/scriptUtils.ts

```diff
@@ -251,7 +251,7 @@ export async function prepareExternalProject(cwd: PortablePath, outputPath: Port
               : [];
 
             // Makes sure that we'll be using Yarn 1.x
-            const version = await execUtils.pipevp(`yarn`, [`set`, `version`, `classic`, `--only-if-needed`], {cwd, env, stdin, stdout, stderr, end: execUtils.EndStrategy.ErrorCode});
+            const version = await execUtils.pipevp(process.execPath, [process.argv[1], `set`, `version`, `classic`, `--only-if-needed`], {cwd, env, stdin, stdout, stderr, end: execUtils.EndStrategy.ErrorCode});
```

Comment:
But doesn't it still change the `packageManager` field, because of [this](https://github.com/yarnpkg/berry/blob/master/packages/plugin-essentials/sources/commands/set/version.ts#L242-L263)?

Reply:
Yes, but that's not what this PR aimed to solve, though I can extend it to handle that as well.